0029-stream/0029-stream.md

```diff
@@ -436,6 +444,24 @@ In other words, if a sender resends data (e.g. because a packet was lost), it MU
 
 Asset details exposed by this frame MUST NOT change during the lifetime of a Connection.
 
+#### 5.3.15. `StreamReceipt` Frame
+
+| Field | Type | Description |
+|---|---|---|
+| Stream ID | VarUInt | Identifier of the stream this frame refers to. |
+| Receipt | 65-Byte OctetString | Proof provided by the receiver of the total amount received on this stream |
+
+The `Receipt` MUST contain the following fields encoded using the [Octet Encoding Rules (OER)](http://www.oss.com/asn1/resources/books-whitepapers-pubs/Overview_of_OER.pdf):
+
+| Field | Type | Description |
+|---|---|---|
+| HMAC | UInt256 | HMAC-SHA256 over all other fields using the 32 byte Receipt Secret, which is pre-shared between the verifying party and the receiver. |
+| Receipt Nonce | UInt128 | A random nonce pre-shared between the verifying party and the receiver used to identify the STREAM connection. |
+| Stream ID | UInt8 | Identifier of the stream this receipt refers to. |
+| Total Received | UInt64 | Total amount, denominated in the units of the receiver, that the receiver has received on this stream thus far. |
+| Stream Start Time | UInt64 | A UNIX timestamp referring to the time that the stream was established at the receiver. |
```

Comment:
This lets the verifier set a cache expiry on how long it remembers the highest amount seen for a Receipt Nonce + Stream Id. If we had no timestamp it would need to remember the highest receipt seen forever or else someone could present their receipt a year later and it would still be valid (and they would be credited the full total received instead of the diff of total received - last total received). So a verifier would basically have some reasonable cache expiry and not accept receipts from >cache expiry ago.

Reply:
Agree you wouldn't want to honor receipts indefinitely, but couldn't the verifier just cache the timestamp it generated the receipt nonce/responded to the SPSP query and use that to expire its cache? Why should it trust the recipient not to manipulate the timestamp?

Reply:
I like the idea of removing the timestamp from the receipt.
Verifiers can either cache the nonce and SPSP query time or encode the time in the nonce if they prefer to be more stateless.

Reply:
:+1: that's a good observation, and saves us from having to specify a timestamp format which I'm a big fan of

Reply:
Did this get resolved?

Reply:
yep, timestamp has been removed https://github.com/interledger/rfcs/pull/570/commits/ff2617a7d1b4ef592426b153d8f10a6b8f0ff180